introduce mvn wrapper

grafjo opened the pull request.

strieflin reviewed (changes requested):
Including the wrapper scripts and jar means we distribute these artifacts. This means, we have to fulfill the license obligations (redistribute license and notice files of the original artifacts). Thus, please create a `3RD-PARTY-LICENSES ` folder in the repository root, add a folder for the Maven wrapper artifacts, and put the license and notice files there. See the client or service modules for examples. 

BTW; You can go ahead and make the required changes for castor (see carbynestack/castor#23) as well as a preparation for the review (FYI: @sbckr).

grafjo commented:
> Including the wrapper scripts and jar means we distribute these artifacts. This means, we have to fulfill the license obligations (redistribute license and notice files of the original artifacts). Thus, please create a `3RD-PARTY-LICENSES ` folder in the repository root, add a folder for the Maven wrapper artifacts, and put the license and notice files there. See the client or service modules for examples.
> 
> BTW; You can go ahead and make the required changes for castor (see [carbynestack/castor#23](https://github.com/carbynestack/castor/pull/23)) as well as a preparation for the review (FYI: @sbckr).

I added the  `3RD-PARTY-LICENSES ` part to this repo and also updated [carbynestack/castor#23]

strieflin commented on mvnw:
```diff
@@ -0,0 +1,316 @@
+  exit 1;
+fi
+
+##########################################################################################
```
I would suggest to remove `maven-wrapper.jar` from the commit. It will be downloaded by the script in case it is not available.

grafjo replied:
good point - I removed the jar

strieflin reviewed (changes requested):
Please remove `sbom.xml`. See carbynestack/castor#23 for a rationale.

Lila84 commented:
> Please remove `sbom.xml`. See [carbynestack/castor#23](https://github.com/carbynestack/castor/pull/23) for a rationale.

Did that.

strieflin reviewed (approved):
LGTM
